### ISSUE TITLE
[Xcode 11] [Catalina] [macOS] [Segmentation fault]

### ISSUE THREAD

**Issue by jedisct1** [jedisct1/libsodium#881]
Just a pinned issue in order to limit the number of duplicate reports we may get now that Catalina is out.

__TL;DR: Xcode 11 generates invalid code. Don't use it.__

On old libsodium versions, the libsodium test suite fails when compiled with Xcode 11 (from beta 5 to Xcode 11 final). 

Xcode 11 is buggy and generates invalid code. This affects more than libsodium - other examples include [OpenZFS](https://github.com/openzfsonosx/zfs/issues/721), oniguruma, ffmpeg, libbluray, x265 and gcc. This may affect your application, and you may not know unless an extensive test suite hits it by accident.

That issue was reported to Apple back in August, when Xcode was still in beta. I independently reported it later on the developer forums:

https://forums.developer.apple.com/thread/121887

If you are using `libsodium-1.0.18-stable` as is has always been [recommended](https://download.libsodium.org/doc/installation), the test suite passes.

However, given that the first releases of Xcode 11 don't seem to have received much testing before being tagged, it is not something we can recommend for any security critical code.

What can you do?

1) Report it to Apple.

2) If you really want/need to use Xcode 11 to compile libsodium, use the `stable` branch or `stable` releases of libsodium. The Xcode bug appeared after the base `1.0.18` version was released, so workarounds for compiler bugs like these go to maintenance releases. In Rust, use the [`libsodium-sys-stable`](https://crates.io/crates/libsodium-sys-stable) crate instead of `libsodium-sys`.

**Comment by jedisct1** [comment]
** Update: this seems to have been fixed in Xcode 11.2 **
